src/components/lab/WavelengthDistribution.js

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { Group, Text, Rect, Line } from 'react-konva';
+import {
+  WAVELENGTH_DISTRIBUTION_HEIGHT,
+  WAVELENGTH_DISTRIBUTION_WIDTH,
+  WAVELENGTH_DISTRIBUTION_STROKE_COLOR,
+  WAVELENGTH_DISTRIBUTION_STROKE_WIDTH,
+  WAVELENGTH_DISTRIBUTION_LABELS_FONT_SIZE,
+  WAVELENGTH_LABELS_Y_AXIS_ADJUSTMENT_FACTOR,
+  WAVELENGTH_DISTRIBUTION_PADDING,
+  WAVELENGTH_DISTRIBUTION_DISTRIBUTION_POINTS_NUMBER,
+  WAVELENGTH_DISTRIBUTION_MIN_WAVELENGTH,
+  WAVELENGTH_DISTRIBUTION_MAX_WAVELENGTH,
+  WAVELENGTH_DISTRIBUTION_MARGIN,
+  WAVELENGTH_DISTRIBUTION_AXIS_FONT_SIZE,
+  WAVELENGTH_DISTRIBUTION_AXIS_HEIGHT,
+  WAVELENGTH_DISTRIBUTION_PADDING_TOP,
+  WAVELENGTH_DISTRIBUTION_PEAK_WAVELENGTH_COLOR,
+  WAVELENGTH_LABELS_X_AXIS_ADJUSTMENT_FACTOR,
+  WAVELENGTH_DISTRIBUTION_AXIS_NB_TICKS,
+  WAVELENGTH_DISTRIBUTION_BACKGROUND_COLOR,
+  WAVELENGTH_DISTRIBUTION_LABELS_PADDING_TOP,
+  SPEED_OF_LIGHT_CONSTANT,
+  PLANCK_CONSTANT,
+  WAVELENGTH_DISTRIBUTION_TICK_LINE_LENGTH,
+} from '../../config/constants';
+
+// source: https://github.com/phetsims/blackbody-spectrum/blob/master/js/blackbody-spectrum/model/BlackbodyBodyModel.js
+const getSpectralPowerDensityAt = ({ wavelength, temperature }) => {
+  // Avoiding division by 0
+  if (wavelength === 0) {
+    return 0;
+  }
+
+  const B = 1.43877e7; // is hc/k in units of nanometer-kelvin
+  const c = SPEED_OF_LIGHT_CONSTANT;
+  const h = PLANCK_CONSTANT;
+  return (
+    (2 * h * c * c) /
+    wavelength ** 5 /
+    (Math.exp(B / (wavelength * temperature)) - 1)
+  );
+};
+
+const WavelengthDistribution = ({ stageDimensions, temperature }) => {
+  const { t } = useTranslation();
+  const { stageHeight, stageWidth } = stageDimensions;
+  const wavelengthDistance =
+    WAVELENGTH_DISTRIBUTION_MAX_WAVELENGTH -
+    WAVELENGTH_DISTRIBUTION_MIN_WAVELENGTH;
+
+  // centers spectrum bar horizontally
+  const wavelengthDistributionInitialXPosition =
+    stageWidth -
+    WAVELENGTH_DISTRIBUTION_WIDTH -
+    WAVELENGTH_DISTRIBUTION_MARGIN -
+    2 * WAVELENGTH_DISTRIBUTION_PADDING;
+  const wavelengthDistributionInitialYPosition = 0.12 * stageHeight;
```

Comment:
Why `0.12`?

Reply:
Because of the header, I'll check if I can rather depend on its height.